src/containers/TemplateBuilder/Templates.tsx

```diff
@@ -153,20 +153,44 @@ const TemplatesWrapper: React.FC = () => (
   </OperationContext>
 )
 
+type SortColumn = 'name' | 'code' | 'category' | 'status'
+
 const Templates: React.FC = () => {
   const { t } = useLanguageProvider()
   const [selectedRow, setSelectedRow] = useState(-1)
   const fileInputRef = useRef<HTMLInputElement>(null)
   const { templates, refetch } = useGetTemplates()
   const { importTemplate } = useOperationState()
+  const [hideInactive, setHideInactive] = useState(false)
+  const [selectedCategories, setSelectedCategories] = useState<string[]>([])
+  const [sortColumn, setSortColumn] = useState<SortColumn>()
+  const [sortAsc, setSortAsc] = useState<1 | -1>(1)
```

Comment:
The reason I've used -1 / 1 rather than `true`/`false` is that you can just multiply it by the result of the compare function to switch the direction. (See "sortTemplates" function)

Reply:
Nice,

I feel like this is on the boundary for me of easy legibility. If I ever were to come back to this sort of thing with fresh eyes, I'd probably take some time to figure it out. Might just be me, though! Looks good.

Reply:
Yeah, fair enough. It's probably on the borderline. I was just happy with how many less lines of calculation it yielded! :)